fastify.js

```diff
@@ -358,6 +360,16 @@ function build (options) {
     return this._contentTypeParser.hasParser(contentType)
   }
 
+  function handleClientError (e, socket) {
+    socket.end(
+      JSON.stringify({
+        error: http.STATUS_CODES['400'],
+        message: 'Client Error',
+        statusCode: 400
+      })
```

Comment:
Yes, using the HTTP protocol, the response body should be conformed with the standard.
But in that callback the second parameter is a tcp socket not a http response.
So ending the socket with a body like this, is better

    const body = JSON.stringify({
      error: http.STATUS_CODES['400'],
      message: 'Client Error',
      statusCode: 400
    })
    socket.end(`HTTP/1.1 400 Bad Request\r\nContent-Length: ${body.length}\r\n\r\n${body}`)